resolvers/node/analyze.rs

```diff
@@ -566,58 +566,68 @@ impl<
     &self,
     entry_specifier: &Url,
     source: Option<Cow<'a, str>>,
+    bundling: bool,
   ) -> Result<Cow<'a, str>, TranslateCjsToEsmError> {
-    let analysis = self
-      .module_export_analyzer
-      .analyze_all_exports(entry_specifier, source)
-      .await?;
-
-    let all_exports = match analysis {
-      ResolvedCjsAnalysis::Esm(source) => return Ok(source),
-      ResolvedCjsAnalysis::Cjs(all_exports) => all_exports,
-    };
-
-    // todo(dsherret): use capacity_builder here to remove all these heap
-    // allocations and make the string writing faster
-    let mut temp_var_count = 0;
-    let mut source = vec![
-      r#"import {createRequire as __internalCreateRequire, Module as __internalModule } from "node:module";
-      const require = __internalCreateRequire(import.meta.url);"#
-        .to_string(),
-    ];
-
-    source.push(format!(
-      r#"let mod;
-      if (import.meta.main) {{
-        mod = __internalModule._load("{0}", null, true)
-      }} else {{
-        mod = require("{0}");
-      }}"#,
-      url_to_file_path(entry_specifier)
-        .unwrap()
-        .to_str()
-        .unwrap()
-        .replace('\\', "\\\\")
-        .replace('\'', "\\\'")
-        .replace('\"', "\\\"")
-    ));
+    let inner = async || {
+      let all_exports = if bundling {
```

Comment:
I'm not sure I'm a fan of passing this boolean all the way down here. Any other ideas how we could do that?

Reply:
David suggested a better solution, so I went with that instead